src/main/java/eu/elixir/ega/ebi/egafuse/filesystems/internal/EgaRemoteFile.java

```diff
@@ -44,35 +43,27 @@
 public class EgaRemoteFile extends EgaApiFile {
 
     private static final long PAGE_SIZE = 1024L * 1024L * 10L;
-    private static final int NUM_PAGES = 7;
-    private final Moshi MOSHI = new Moshi.Builder().build();
+    private static int NUM_PAGES;
     private EgaFileDto theFile;
     private LoadingCache<Integer, byte[]> cache;
 
     public EgaRemoteFile(String name, EgaApiDirectory parent) {
         super(name, parent);
         setType();
-
-        // Init cache
-        this.cache = CacheBuilder.newBuilder()
-                .maximumSize(NUM_PAGES)
-                .concurrencyLevel(NUM_PAGES * 2)
-                .build(
-                        new CacheLoader<Integer, byte[]>() {
-                            public byte[] load(Integer page) throws Exception {
-                                return populateCache(page);
-                            }
-                        });
+        setCache();
     }
 
     public EgaRemoteFile(String name, EgaApiDirectory parent, EgaFileDto theFile) {
         super(name, parent);
         this.theFile = theFile;
         setType();
+        setCache();
+    }
 
-        // Init cache
+    private void setCache() {
```

Comment:
This method is not a setter, it is a whole initialization, please change name accordingly

Reply:
> Can you clarify me one thing, is that cache per file?

Yes cache is made per file

Reply:
> This method is not a setter, it is a whole initialization, please change name accordingly

Okay I will change the name

Reply:
> > Can you clarify me one thing, is that cache per file?
> 
> Yes cache is made per file

Then I'm sorry but this approach will not work. The way it is implemented you will have N connections at most for a specific file but that means that if they read multiple files we are gonna have potentially too many connections per instance going against our system without any bound. We will need to decide in an architecture approach and redo how it has been done.